testbed/testbed/test_case.go

```diff
@@ -34,10 +34,9 @@ type TestCase struct {
 	// Agent process.
 	agentProc OtelcolRunner
 
-	Sender   DataSender
```

Comment:
I think this is the main breaking change, with `Sender` now only being available on the `ProviderSender` instance, which is still available on the `LoadGenerator` field. In the cases where this was used, the tests have already access to the sender directly so the indirect access is unnecessary to begin with. I decided to not address this directly in this initial pass to keep the changes localized to the involved types.